gc/base/GCExtensionsBase.hpp

```diff
@@ -832,6 +833,16 @@ class MM_GCExtensionsBase : public MM_BaseVirtual {
 #endif /* defined(OMR_GC_MODRON_SCAVENGER) */
 	}
 
+	MMINLINE bool
+	isEvacuatorEnabled()
+	{
+#if defined(OMR_GC_MODRON_SCAVENGER)
+		return evacuatorEnabled && isScavengerEnabled() && !isConcurrentScavengerEnabled();
```

Comment:
can the evacuator be enabled and not the scavenger?  I think it would be better if this code looked like this:
```
Assert_MM_true(isScavengerEnabled());
Assert_MM_true(!isConcurrentScavengerEnabled());
return evacuatorEnabled;
```

Reply:
I agree and will change this as you suggest.

Reply:
Unfortunately ModronAssertions.h cannot be included in GCExtensionsBase.hpp without compilation errors (redefined macro, unrelated to current PR changes). Leaving this test as is will simply force scavenger to be selected for generational GC if concurrent scavenger is enabled, without warning as long as scavenger is enabled, and evacuator enablement is effective only when gencon is selected as gc policy (because evacuator is implemented in MM_Scavenger, which is only instantiated for gencon). I don't know that printing an error message in this context is warranted or would be useful.